kernel/src/main/java/org/kframework/compile/checks/CheckAtt.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2016-2019 K Team. All Rights Reserved.
+package org.kframework.compile.checks;
+
+import org.kframework.attributes.Att;
+import org.kframework.attributes.HasLocation;
+import org.kframework.definition.Module;
+import org.kframework.definition.Rule;
+import org.kframework.definition.Sentence;
+import org.kframework.utils.errorsystem.KEMException;
+
+import java.util.Set;
+
+/**
+ * Created by dwightguth on 1/25/16.
+ */
+public class CheckAtt {
+    private final Set<KEMException> errors;
+    private final Module m;
+
+    public CheckAtt(Set<KEMException> errors, Module m) {
+        this.errors = errors;
+        this.m = m;
+    }
+
+    public void check(Sentence sentence) {
+        if (sentence instanceof Rule) {
+            check(((Rule) sentence).att(), sentence);
+        }
+    }
+
+    private void check(Att att, HasLocation loc) {
+        if (att.contains(Att.OWISE()) && att.contains(Att.SIMPLIFICATION())) {
+          errors.add(KEMException.compilerError("owise attribute is not supported on simplification rules.", loc));
+        }
+        if (att.contains(Att.PRIORITY()) && att.contains(Att.SIMPLIFICATION())) {
+          errors.add(KEMException.compilerError("priority attribute is not supported on simplification rules.", loc));
+        }
```

Comment:
We should allow priority on simplification rules. For example, this is needed for the arithmetic simplification lemmas.

Reply:
@ana-pantilie could you provide an example?

Reply:
@traiansf [These](https://github.com/kframework/k/blob/a34d4d155bb4bfa58194a892f36845b411f373b1/k-distribution/include/kframework/builtin/domains.k#L451) rules need to apply after any of the less general rules below.

Reply:
@ana-pantilie I agree it makes sense to try some simplification rules only after other simplification rules have been attempted, but I'm wondering whether using `priority` would not create confusion.
The reason I'm saying this is that for other cases (rules, function definitions), priority has semantic meaning (forces a check that rules with less priority cannot be applies), while priority for simplification axioms probably shouldn't, as it is more of a preference to attempt it before other simplification axioms. Maybe we could allow the `simplification` attribute to take an extra argument?

Reply:
@traiansf Yes, you're right, it could be confusing. @ttuegel what do you think?

Reply:
> Maybe we could allow the simplification attribute to take an extra argument?

That sounds fine.

Reply:
@dwightguth I'm trying to add an optional argument to the `simplification` attribute so we can write (for example) `simplification(51)`. When I add arguments like that, the frontend starts to encode `simplification` (without arguments) as `simplification("")` in the generated Kore. I want similar behavior to the `concrete` attribute: `concrete` becomes `concrete()`, `concrete(X)` becomes `concrete("X")`, etc. Can you advise how I might do that?

Reply:
I have no idea how that happens the way it does now with the concrete attribute. If anything that sounds suspicious to me and indicates that the code is probably doing something unintended. The way it is supposed to work in the current code is to decide whether the attribute has an argument based on whether a nonempty string appears anywhere as the argument of the attribute. So in that sense what you describe is the intended behavior. The k frontend doesn't actually know how to distinguish between an attribute `foo` and an attribute `foo()`. This could in theory be changed, but it's not remotely trivial and would likely be a very intrusive change to the frontend.